src/data/polynomial/derivative.lean

```diff
@@ -63,38 +77,32 @@ derivative_C
 
 @[simp] lemma derivative_add {f g : polynomial R} :
   derivative (f + g) = derivative f + derivative g :=
-by refine finsupp.sum_add_index _ _; intros;
-simp only [add_mul, zero_mul, C_0, C_add, C_mul]
-
-/-- The formal derivative of polynomials, as additive homomorphism. -/
-def derivative_hom (R : Type*) [semiring R] : polynomial R →+ polynomial R :=
-{ to_fun := derivative,
-  map_zero' := derivative_zero,
-  map_add' := λ p q, derivative_add }
+derivative.map_add f g
 
 @[simp] lemma derivative_neg {R : Type*} [ring R] (f : polynomial R) :
-  derivative (-f) = -derivative f :=
-(derivative_hom R).map_neg f
+  derivative (-f) = - derivative f :=
+linear_map.map_neg derivative f
```

Comment:
We decided in the other PR that this didn't work?
```suggestion
  derivative (-f) = - derivative f :=
derivative.map_neg f
```

Reply:
The few lemmas I have left as `linear_map.map_foo derivative` don't compile as `derivative.map_foo`, at least on my computer.

Reply:
Weird... anyway, I guess that shouldn't be fixed in this PR.